src/app/app/rooms/page.js

```diff
@@ -0,0 +1,19 @@
+import CreateRoomBox from "@/app/_components/CreateRoomBox/CreateRoomBox.js";
+import RoomsOverview from "@/app/_components/RoomsOverview.js";
+import TopBar from "@/app/_components/TopBar/TopBar.js";
+import "@/app/globals.css";
+import { getProfile } from "@/utils/api";
+
+export default async function HomePage() {
+  const user = await getProfile();
+
+  return (
+    <div>
+      <TopBar title={"Rooms"} />
+      <div className="mainContainer">
+        <RoomsOverview rooms={user.rooms} />
```

Comment:
this breaks the code

⨯ TypeError: Cannot read properties of null (reading 'rooms')
    at LandinPage (./src/app/page.js:33:37)
  12 |       <TopBar title={"Rooms"} />
  13 |       <div className="mainContainer">
> 14 |         <RoomsOverview rooms={user.rooms} />
     |                                   ^
  15 |         <CreateRoomBox context={{ user: user }} />
  16 |       </div>
  17 |     </div>
 ⨯ src/app/page.js (14:35) @ rooms
 ⨯ TypeError: Cannot read properties of null (reading 'rooms')
    at LandinPage (./src/app/page.js:33:37)
  12 |       <TopBar title={"Rooms"} />
  13 |       <div className="mainContainer">
> 14 |         <RoomsOverview rooms={user.rooms} />
     |                                   ^
  15 |         <CreateRoomBox context={{ user: user }} />
  16 |       </div>
  17 |     </div>

Reply:
It breaks the first time you sign up because the profile isn't associated with any rooms yet. We can load this if rooms != null or load a view to create your first room if user.rooms.length == 0?

Reply:
Actually, I fixed this and also now we redirect the user to create a profile right after they sign up